src/darwin/agent/reportcrash.js

```diff
@@ -96,45 +97,70 @@ Interceptor.attach(Module.getExportByName(CORESYMBOLICATION_PATH, 'task_is_64bit
     this.pid = pidForTask(args[0].toUInt32());
   },
   onLeave: function (retval) {
-    if (this.pid === crashedPid)
-      is64Bit = !!retval.toUInt32();
+    const localState = state[this.threadId];
+    if (localState === undefined) {
+      console.error('task_is_64bit: missing state for thread', this.threadId);
+      return;
+    }
+    if (this.pid === localState.crashedPid)
+      localState.is64Bit = !!retval.toUInt32();
   }
 });
 
 Interceptor.attach(CrashReport['- isActionable'].implementation, {
   onLeave: function (retval) {
     var isActionable = !!retval.toInt32();
+    const localState = state[this.threadId];
+    if (localState === undefined) {
+      console.error('- isActionable: missing state for thread', this.threadId);
+      return;
+    }
     if (!isActionable) {
       retval.replace(ptr(1));
-      forcedByUs = true;
+      localState.forcedByUs = true;
     }
   },
 });
 
 Interceptor.attach(NSMutableDictionary['- logCounter_isLog:byKey:count:withinLimit:withOptions:'].implementation, {
   onLeave: function (retval) {
     var isLogWithinLimit = !!retval.toInt32();
+    const localState = state[this.threadId];
+    if (localState === undefined) {
+      console.error('- logCounter_isLog: missing state for thread', this.threadId);
+      return;
+    }
     if (!isLogWithinLimit) {
       retval.replace(ptr(1));
-      forcedByUs = true;
+      localState.forcedByUs = true;
     }
   },
 });
 
 Interceptor.attach(Module.getExportByName(LIBSYSTEM_KERNEL_PATH, 'rename'), {
   onEnter: function (args) {
     var newPath = args[1].readUtf8String();
+    const localState = state[this.threadId];
+    if (localState === undefined) {
+      console.error('rename: missing state for thread', this.threadId);
+      return;
+    }
     if (/\.ips$/.test(newPath)) {
-      logPath = newPath;
+      localState.logPath = newPath;
     }
   },
 });
 
 Interceptor.attach(AppleErrorReport['- saveToDir:'].implementation, {
   onLeave: function (retval) {
-    if (forcedByUs) {
-      unlink(Memory.allocUtf8String(logPath));
-      reset();
+    const localState = state[this.threadId];
+    if (localState === undefined) {
+      console.error('- saveToDir: missing state for thread', this.threadId);
+      return;
+    }
+    if (localState.forcedByUs) {
+      unlink(Memory.allocUtf8String(localState.logPath));
+      delete state[this.threadId];
```

Comment:
i wonder if it's better to move this outside the `if`

Reply:
Yep I think that's a good idea.